packages/ra-ui-materialui/src/field/ReferenceOneField.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ReactElement } from 'react';
+import PropTypes from 'prop-types';
+import { Typography } from '@mui/material';
+import {
+    useReferenceOneFieldController,
+    useRecordContext,
+    ResourceContextProvider,
+    getResourceLinkPath,
+    LinkToType,
+} from 'ra-core';
+
+import { PublicFieldProps, InjectedFieldProps } from './types';
+import { ReferenceFieldView } from './ReferenceField';
+
+/**
+ * Render the related record in a one-to-one relationship
+ *
+ * Expects a single field as child
+ *
+ * @example // display the bio of the current author
+ * <ReferenceOneField reference="bios" target="author_id">
+ *     <TextField source="body" />
+ * </ReferenceOneField>
+ */
+export const ReferenceOneField = (props: ReferenceOneFieldProps) => {
+    const {
+        basePath,
+        resource,
+        children,
+        reference,
+        source,
+        target,
+        emptyText,
+        link = false,
+    } = props;
+    const record = useRecordContext(props);
+
+    const {
+        isLoading,
+        isFetching,
+        referenceRecord,
+        error,
+        refetch,
+    } = useReferenceOneFieldController({
+        record,
+        reference,
+        source,
+        target,
+    });
+
+    const resourceLinkPath = getResourceLinkPath({
+        basePath,
+        link,
+        record,
+        reference,
+        resource,
+        source,
+    });
+
+    return !record || (!isLoading && referenceRecord == null) ? (
+        emptyText ? (
+            <Typography component="span" variant="body2">
+                {emptyText}
+            </Typography>
+        ) : null
+    ) : (
+        <ResourceContextProvider value={reference}>
+            <ReferenceFieldView
+                isLoading={isLoading}
```

Comment:
I get a typescript error here: 
`Property 'isLoading' does not exist on type 'IntrinsicAttributes & ReferenceFieldViewProps & { children?: ReactNode; }'.`

Reply:
same